src/simsopt/field/biotsavart.py

```diff
@@ -2,6 +2,53 @@
 import simsoptpp as sopp
 from simsopt.field.magneticfield import MagneticField
 from simsopt.geo.curve import Curve
+from simsopt._core.graph_optimizable import Optimizable
+from simsopt._core.derivative import Derivative
+
+
+class Coil(sopp.Coil, Optimizable):
+
+    def __init__(self, curve, current):
+        self.__curve = curve
+        self.__current = current
+        sopp.Coil.__init__(self, curve, current)
+        Optimizable.__init__(self, x0=np.asarray([]), depends_on=[curve, current])
+
+    def vjp(self, v_gamma, v_gammadash, v_current):
+        return self.curve.dgamma_by_dcoeff_vjp(v_gamma) \
+            + self.curve.dgammadash_by_dcoeff_vjp(v_gammadash) \
+            + self.current.vjp(v_current)
+
+
+class Current(sopp.Current, Optimizable):
+
+    def __init__(self, current):
+        sopp.Current.__init__(self, current)
+        Optimizable.__init__(self, external_dof_setter=Current.set_dofs,
+                             x0=self.get_dofs())
+
+    def set_dofs(self, dofs):
+        sopp.Current.set_dofs(self, dofs)
+
```

Comment:
Here `set_dofs` is just passing dofs to `sopp.Current.set_dofs`. Can't we just use `sopp.Current.set_dofs` in line 27 and get rid of this method?

Reply:
Yep, good point.